src/analysis/retail/priest/shadow/modules/talents/Archon/ResonantEnergy.tsx

```diff
@@ -0,0 +1,57 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { Options } from 'parser/core/Module';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import { calculateEffectiveDamage } from 'parser/core/EventCalculateLib';
+import Enemies from 'parser/shared/modules/Enemies';
+
+import TALENTS from 'common/TALENTS/priest';
+import SPELLS from 'common/SPELLS';
+
+import Events, { DamageEvent } from 'parser/core/Events';
+
+import { ARCHON_RESONANT_ENERGY_MULTIPLIER } from '../../../constants';
+
+class ResonantEnergy extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  protected enemies!: Enemies;
+
+  damageResonantEnergy = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.RESONANT_ENERGY_TALENT);
+
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.checkAmpDamage);
+  }
+
+  checkAmpDamage(event: DamageEvent) {
+    //Damage from Resonant Energy
+    const target = this.enemies.getEntity(event);
+    if (!target) {
+      return;
+    }
+    const ampRE =
+      target.getBuffStacks(SPELLS.SHADOW_PRIEST_ARCHON_RESONANT_ENERGY_DEBUFF.id) *
+      ARCHON_RESONANT_ENERGY_MULTIPLIER;
```

Comment:
this is probably accidentally quadratic, but it might be okay if Resonant Energy doesn't trigger very often on each target. if you encounter performance issues, this would be something to potentially look at.

Reply:
I haven't noticed any performance issues even in some keys, but I've noted this potential issue in the code in case it does become a problem.  If there is another way to calculate this damage amp that doesn't have that potential issue I can change it before it becomes a problem.